h/views/api_auth.py

```diff
@@ -2,102 +2,88 @@
-        page.
-        """
-        self._check_params()
+        scopes, credentials = self.oauth.validate_authorization_request(self.request.url)
 
         if self.request.authenticated_userid is None:
             raise HTTPFound(self.request.route_url('login', _query={
                               'next': self.request.url}))
 
-        params = self.request.params
+        client_id = credentials.get('client_id')
+        state = credentials.get('state')
+
         user = self.user_svc.fetch(self.request.authenticated_userid)
+        client = self.request.db.query(models.AuthClient).get(client_id)
 
         return {'username': user.username,
-                'client_name': 'Hypothesis',
-                'client_id': params['client_id'],
-                'response_type': params['response_type'],
-                'response_mode': params['response_mode'],
-                'state': params.get('state')}
+                'client_name': client.name,
+                'client_id': client.id,
+                'response_type': client.response_type.value,
+                'state': state}
 
     @view_config(request_method='POST',
-                 renderer='h:templates/oauth/post_authorize.html.jinja2',
                  effective_principals=security.Authenticated)
     def post(self):
-        """
-        Process an authentication request and return an auth code to the client.
-
-        Depending on the "response_mode" parameter the auth code will be
-        delivered either via a redirect or via a `postMessage` call to the
-        opening window.
-        """
-        authclient = self._check_params()
-
+        # We don't support scopes at the moment, but oauthlib does need a scope,
+        # so we're adding a hard-coded `annotation:read` and `annotation:write` scopes.
+        scopes = ['annotation:read', 'annotation:write']
         user = self.user_svc.fetch(self.request.authenticated_userid)
+        credentials = {'user': user}
 
-        # Create an "authorization code" for the response.
-        # This is in fact just a JWT grant token since auth codes have not yet
-        # been implemented.
-        auth_code = self.oauth_svc.create_grant_token(user, authclient)
-
-        params = self.request.params
-
-        return {'code': auth_code,
-                # Once authclients have an Origin property, that should be used
-                # instead here.
-                'origin': self.request.host_url,
-                'state': params.get('state')}
+        headers, body, status = self.oauth.create_authorization_response(
+                self.request.url, scopes=scopes, credentials=credentials)
 
-    def _check_params(self):
-        """
-        Check parameters for the authorization request.
+        if 'Location' in headers:
+            return exception_response(status, location=headers['Location'])
+        else:
+            log.error('Created OAuth 2 authorization code, but got no redirect location')
```

Comment:
Given that it seems like this should be a truly exceptional circumstance, it would make more sense to me if this were a `try: ... except: ...`, e.g.

```python
try:
    return exception_response(status, location=headers['Location'])
except KeyError:
    raise RuntimeError('created authorisation code for client "{}" but got no redirect location'.format(...)) 
```

Note that this way you'll get some useful debugging information in Sentry. `raise HTTPInternalServerError()` is going to result in a very bizarre looking error report.